packages/slonik/src/factories/createConnectionPool.ts

```diff
@@ -144,22 +144,25 @@ export const createConnectionPool = ({
       pendingConnections.push(pendingConnection);
 
       const connection = await pendingConnection.catch((error) => {
-        pendingConnections.pop();
+        pendingConnections.splice(
+          pendingConnections.indexOf(pendingConnection),
+          1,
+        );
 
         throw error;
       });
 
       const onRelease = () => {
+        if (connection.state() !== 'IDLE') {
```

Comment:
I moved this up further. If not technically we might call `waitingClients.shift()` but then never actually acquire a connection for that waiting client since we throw an error.

Reply:
This is causing some tests to fail because it's throwing `Connection is not idle.`. From logging it looks like the state is actually `PENDING_DESTROY`. Which I think indicates that this is a true problem in practice, we shouldn't be calling `waitingClients.shift()` if we can't actually do something with that waiting client.

Maybe we should change this to:
```
const onRelease = () => {
        if (connection.state() !== 'IDLE') {
            return;
        }

        ...
```
?